### ISSUE TITLE
Patch to fix buffer overflow

### ISSUE THREAD

**Issue by doublex** [libssh2/libssh2#141]
``` diff
*** userauth.c-old      2016-10-25 08:43:39.000000000 +0200
--- userauth.c  2016-10-26 12:12:22.921837935 +0200
*************** memory_read_publickey(LIBSSH2_SESSION *
*** 487,493 ****

      sp1++;

!     if ((sp2 = memchr(sp1, ' ', pubkey_len - (sp1 - pubkey - 1))) == NULL) {
          /* Assume that the id string is missing, but that it's okay */
          sp2 = pubkey + pubkey_len;
      }
--- 487,493 ----

      sp1++;

!     if ((sp2 = memchr(sp1, ' ', pubkey_len - (sp1 - pubkey + 1))) == NULL) {
          /* Assume that the id string is missing, but that it's okay */
          sp2 = pubkey + pubkey_len;
      }
```

**Comment by doublex** [comment]
Maybe it would be better to write:
````
!     if ((sp2 = memchr(sp1, ' ', pubkey_len - (sp1 - pubkey))) == NULL) {
````

**Comment by alamaison** [comment]
Could you open a Pull Request with this patch?  It makes it hard to review like this and even harder to merge

**Comment by willco007** [comment]
fixed with #159
